02_knight_and_princess/tevfik_aksoy/main.cpp

```diff
@@ -0,0 +1,147 @@
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <set>
+#include <string>
+#include <tuple>
+#include <vector>
+
+using std::array;
+using std::endl;
+using std::get;
+using std::to_string;
+using std::vector;
+
+using VertexId = int;
+using EdgeId = int;
+
+// Tuple of <int,int> for realization of vector of vectors of <int,int>
+using dest_id_edge_id = std::tuple<int, int>;
+
+struct Vertex {
+ public:
+  VertexId id = 0;

```

Comment:
`0` - это валидный id, а так как у нас id должен создаваться и назначаться в конструкторе, было бы лучше указывать стандартным какое-то невалидное значение, которое лучше вынести в константу:
```cpp
constexpr int INVALID_ID = -1;
```
```suggestion
  VertexId id = INVALID_ID;
```

Reply:
Понял, так даже красивее. И вопрос, так понимаю что конст значение лучше глобальным объявлять, да? Исправляю